api/base_controller.go

```diff
@@ -187,34 +189,61 @@ func (c *BaseController) GetSingleObject(r *web.Request) (*web.Response, error)
 // ListObjects handles the fetching of all objects
 func (c *BaseController) ListObjects(r *web.Request) (*web.Response, error) {
 	ctx := r.Context()
-	if ctx.Value("limit").(int) == 0 {
-		log.C(ctx).Debugf("Getting count of %s since max_items is 0", c.objectType)
-		count, err := c.repository.Count(ctx, c.objectType, query.CriteriaForContext(ctx)...)
-		if err != nil {
-			return nil, util.HandleStorageError(err, string(c.objectType))
-		}
+	userCriteria := query.UserCriteriaForContext(ctx)
+	count, err := c.repository.Count(ctx, c.objectType, userCriteria...)
```

Comment:
By the time the objects are listed, the count may have changed. Describe this as a limitation in the document